create_base_image/create_base_image_hostlib.sh

```diff
@@ -0,0 +1,222 @@
+    FLAGS_dest_image="halyard-${CF_VER}-${FLAGS_build_branch}-${FLAGS_build_target}-${FLAGS_build_id}"
+  fi
+  if [[ -z "${FLAGS_dest_family}" ]]; then
+    FLAGS_dest_family="halyard-${FLAGS_build_branch}-${FLAGS_build_target}"
+  fi
+
+  if [[ -n "${FLAGS_dest_family}" ]]; then
+    dest_family_flag=("--family=${FLAGS_dest_family}")
+  else
+    dest_family_flag=()
+  fi
+
+  # Deletes instances, disks and images with names that will be used for build
+  delete_instances=("${FLAGS_build_instance}" "${FLAGS_dest_image}")
+  gcloud compute instances delete -q \
+    "${PZ[@]}" "${delete_instances[@]}" || echo Not running
+  gcloud compute disks delete -q \
+    "${PZ[@]}" "${FLAGS_dest_image}" || echo No scratch disk
+  gcloud compute images delete -q \
+    --project="${FLAGS_build_project}" "${FLAGS_dest_image}"  || echo Not respinning
+
+
+  # BUILD INSTANCE AND DISK CREATION
+
+  gcloud compute disks create \
+    "${PZ[@]}" \
+    --image-family="${FLAGS_source_image_family}" \
+    --image-project="${FLAGS_source_image_project}" \
+    --size=30GB \
+    "${FLAGS_dest_image}"
+
+
+  # Checks if gpu available 
+  local gpu_type="nvidia-tesla-p100-vws"
+  gcloud compute accelerator-types describe "${gpu_type}" "${PZ[@]}" || \
+    fatal_echo "Please use a zone with ${gpu_type} GPUs available."
+
+  gcloud compute instances create \
+    "${PZ[@]}" \
+    --machine-type=n1-standard-16 \
+    --image-family="${FLAGS_source_image_family}" \
+    --image-project="${FLAGS_source_image_project}" \
+    --boot-disk-size=200GiB \
+    --accelerator="type=${gpu_type},count=1" \
+    --maintenance-policy=TERMINATE \
+    "${build_tags[@]}" \
+    "${FLAGS_build_instance}"
+
+  # Wait until instance is booted
+  wait_for_instance "${PZ[@]}" "${FLAGS_build_instance}"
+
+  gcloud compute instances attach-disk \
+      "${PZ[@]}" "${FLAGS_build_instance}" --disk="${FLAGS_dest_image}"
+
+  gcloud compute scp "${PZ[@]}" -- -r \
+    "${source_files[@]}" \
+    "${FLAGS_build_instance}:"
+
+
+  # IMAGE CREATION
+
+  gcloud compute ssh "${PZ[@]}" "${FLAGS_build_instance}" -- \
+    ./create_base_image_gce.sh
+
+  gcloud compute instances delete -q \
+    "${PZ[@]}" "${FLAGS_build_instance}"
+
+  gcloud compute images create \
+    --project="${FLAGS_build_project}" \
+    --source-disk="${FLAGS_dest_image}" \
+    --source-disk-zone="${FLAGS_build_zone}" \
+    --licenses=https://www.googleapis.com/compute/v1/projects/vm-options/global/licenses/enable-vmx \
+    "${dest_family_flag[@]}" \
+    "${FLAGS_dest_image}"
+
+  gcloud compute disks delete -q "${PZ[@]}" \
+    "${FLAGS_dest_image}"
+  
+  cat <<EOF
+    echo Test and if this looks good, consider releasing it via:
+
+    gcloud compute images create \
+      --project="${FLAGS_dest_project}" \
+      --source-image="${FLAGS_dest_image}" \
+      --source-image-project="${FLAGS_build_project}" \
+      "${dest_family_flag[@]}" \
+      "${FLAGS_dest_image}"
+EOF
```

Comment:
This message is unnecessary, if everything goes well you should add the image to the family unconditionally.

Reply:
True, I'll just build the image in `FLAGS_dest_project` directly instead

Reply:
No, keep the destination project separate from the building project (the destination project could be a publicly accessible one, for production machines, while the project where the image is being built could be internal only).
What I meant is that you should run that command instead of printing it. Basically, just remove the cat and the echo, just leave the gcloud invocation

Reply:
Ok 👌

Reply:
I forgot to mention that should only run the command if the dest_project is not empty and different than the build project. It should be inside an 'if' block

Reply:
Ok! If it is not different than the build project then it just gives a gcloud warning but it doesn't break execution. I'll check it anyways